src/lib/address_resolve/AddressResolve_DefaultImpl.cpp

```diff
@@ -30,41 +30,28 @@ void NodeLookupHandle::ResetForLookup(System::Clock::Timestamp now, const NodeLo
 {
     mRequestStartTime = now;
     mRequest          = request;
-    mBestResult       = ResolveResult();
-    mBestAddressScore = Dnssd::IPAddressSorter::IpScore::kInvalid;
+    mResults          = NodeLookupResults();
 }
 
 void NodeLookupHandle::LookupResult(const ResolveResult & result)
 {
 #if CHIP_PROGRESS_LOGGING
     char addr_string[Transport::PeerAddress::kMaxToStringSize];
     result.address.ToString(addr_string);
-#endif
-
-    auto newScore = Dnssd::IPAddressSorter::ScoreIpAddress(result.address.GetIPAddress(), result.address.GetInterface());
-    if (to_underlying(newScore) > to_underlying(mBestAddressScore))
-    {
-        mBestResult       = result;
-        mBestAddressScore = newScore;
 
-        if (!mBestResult.address.GetIPAddress().IsIPv6LinkLocal())
-        {
-            // Only use the DNS-SD resolution's InterfaceID for addresses that are IPv6 LLA.
-            // For all other addresses, we should rely on the device's routing table to route messages sent.
-            // Forcing messages down an InterfaceId might fail. For example, in bridged networks like Thread,
-            // mDNS advertisements are not usually received on the same interface the peer is reachable on.
-            mBestResult.address.SetInterface(Inet::InterfaceId::Null());
-            ChipLogDetail(Discovery, "Lookup clearing interface for non LL address");
-        }
+    auto score = Dnssd::IPAddressSorter::ScoreIpAddress(result.address.GetIPAddress(), result.address.GetInterface());
```

Comment:
UpdateResult below immediately calls `ScoreIpAddress`. Suggest a way that doesn't force the score to be computed twice